x-pack/plugin/core/src/main/java/org/elasticsearch/datastreams/logsdb/LogsdbIndexModeSettingsProvider.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.datastreams.logsdb;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.metadata.ComposableIndexTemplate;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.metadata.MetadataIndexTemplateService;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.index.IndexSettingProvider;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class LogsdbIndexModeSettingsProvider implements IndexSettingProvider {
+    private static final Logger logger = LogManager.getLogger(LogsdbIndexModeSettingsProvider.class);
+
+    public static final Setting<Boolean> CLUSTER_LOGSDB_ENABLED = Setting.boolSetting(
+        "cluster.logsdb.enabled",
+        false,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final String LOGS_PATTERN = "logs-.+-.+";
```

Comment:
Since this is used in the context of `Regex.simpleMatch`, I think this needs to be adapted
```suggestion
    public static final String LOGS_PATTERN = "logs-*-*";
```